library_registry/admin/routes.py

```diff
@@ -10,59 +14,91 @@
     'admin', __name__,
     template_folder='templates')
 
+# Using an `after_request` callback, we refresh any token that is within 30
+# minutes of expiring. Change the timedeltas to match the needs of your application.
+
+
+@admin.after_request
+def refresh_expiring_jwts(response):
+    try:
+        exp_timestamp = get_jwt()["exp"]
+        now = datetime.now(timezone.utc)
+        target_timestamp = datetime.timestamp(now + timedelta(minutes=30))
+        if target_timestamp > exp_timestamp:
+            access_token = create_access_token(identity=get_jwt_identity())
+            response = make_response(response, 201)
+            set_access_cookies(response, access_token)
+        return response
+    except (RuntimeError, KeyError):
+        # Case where there is not a valid JWT. Just return the original response
+        return response
+
+
 @admin.route('/admin/', strict_slashes=False)
 def admin_view():
     return current_app.library_registry.view_controller()
 
+
 @admin.route('/admin/log_in', methods=["POST"])
+@admin.route('/admin/log_in/<jwt_preferred>')
 @returns_problem_detail
-def log_in():
-    return current_app.library_registry.admin_controller.log_in()
+def log_in(jwt_preferred=False):
+    return current_app.library_registry.admin_controller.log_in(jwt_preferred)
+
 
 @admin.route('/admin/log_out')
 @check_logged_in
 @returns_problem_detail
+@jwt_required(optional=True)
```

Comment:
With the new combined approach above this probably isn't required, right?

Reply:
Missed the duplication, thanks.